torchesn/nn/reservoir.py

```diff
@@ -80,7 +80,7 @@ def reset_parameters(self):
                                                    1 - self.density))]
                     w_hh[zero_weights] = 0
                 w_hh = w_hh.view(self.hidden_size, self.hidden_size)
-                abs_eigs = (torch.eig(w_hh)[0] ** 2).sum(1).sqrt()
+                abs_eigs = torch.abs(torch.linalg.eigvals(w_hh))
```

Comment:
not sure about using `torch.abs`, I remember that it didn't support complex numbers. Can you check it out? Otherwise, just replace the line with:

`abs_eigs = (torch.linalg.eig(w_hh)[0] ** 2).sum(1).sqrt()`

Reply:
We may use `eigvals` instead of `eig` as well, but I'm not sure if we would need to change the dimension on `sum(1)` and I cannot test it right now.

Reply:
Oh nice, we're good to go then, thank you for the contribution!